apps/gp2-server/src/controllers/output.controller.ts

```diff
@@ -0,0 +1,233 @@
+import { NotFoundError } from '@asap-hub/errors';
+import {
+  AuthorPostRequest,
+  AuthorUpsertDataObject,
+  gp2 as gp2Model,
+  ValidationErrorResponse,
+  VALIDATION_ERROR_MESSAGE,
+} from '@asap-hub/model';
+import Boom from '@hapi/boom';
+import { ExternalAuthorDataProvider } from '../data-providers/external-authors.data-provider';
+import { OutputDataProvider } from '../data-providers/output.data-provider';
+
+export default class Outputs implements OutputController {
+  constructor(
+    private outputDataProvider: OutputDataProvider,
+    private externalAuthorDataProvider: ExternalAuthorDataProvider,
+  ) {}
+
+  async fetchById(outputId: string): Promise<gp2Model.OutputResponse> {
+    const output = await this.outputDataProvider.fetchById(outputId);
+    if (!output) {
+      throw new NotFoundError(
+        undefined,
+        `output with id ${outputId} not found`,
+      );
+    }
+
+    return output;
+  }
+
+  async fetch(options: FetchOptions): Promise<gp2Model.ListOutputResponse> {
+    const { filter: fetchFilter, ...fetchOptions } = options;
+
+    const filter: gp2Model.FetchOutputOptions['filter'] = Array.isArray(
+      fetchFilter,
+    )
+      ? { documentType: fetchFilter }
```

Comment:
I'm a bit confused :D

Reply:
Copied from the research output controller: https://github.com/yldio/asap-hub/blob/a275282553687f40e935bf158c07c805d9384b35/apps/crn-server/src/controllers/research-outputs.ts#L61-L65

It confused me too, the filter can take a string[]. If it does, then it's an array of documentTypes.